src/late_multicellular_stage/MetaballLayout.cs

```diff
@@ -209,4 +210,26 @@ public IEnumerable<T> GetMetaballsNotTouchingParents(float contactThreshold = 0.
             }
         }
     }
+
+    public Metaball? GetClosestMetaballToPosition(Vector3 position)
+    {
+        var closestDistance = 10000.0f;
+        Metaball? closestMetaball = null;
+
+        foreach (var metaball in this)
+        {
+            if (metaball.Parent == null)
+                continue;
+
+            var distance = metaball.Position.DistanceTo(position);
```

Comment:
To save massively on performance this should use squared distances. (as calculating the distance requires taking a square root).